pkg/maps/nat/cell.go

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package nat
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/promise"
+	"github.com/cilium/cilium/pkg/tuple"
+	"github.com/cilium/hive/cell"
+)
+
+// MapDisabled is the expected error will be if map was not created
+// due to configuration.
+var MapDisabled = fmt.Errorf("nat map is disabled")
+
+// Cell exposes global nat maps via Hive. These maps depend on
+// the final state of EnableNodePort, thus the maps are currently
+// provided as promises.
+// TODO: Once we have a way of finalizing this config prior to runtime
+// we'll want to provide these using bpf.MapOut[T] (GH: #32557)
+var Cell = cell.Module(
+	"nat-maps",
+	"NAT Maps",
+	cell.Provide(func(lc cell.Lifecycle, cfgPromise promise.Promise[*option.DaemonConfig]) (promise.Promise[NatMap4], promise.Promise[NatMap6]) {
+		var ipv4Nat, ipv6Nat *Map
+		res4, promise4 := promise.New[NatMap4]()
+		res6, promise6 := promise.New[NatMap6]()
+
+		lc.Append(cell.Hook{
+			OnStart: func(hc cell.HookContext) error {
+				ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
+				defer cancel()
+				cfg, err := cfgPromise.Await(ctx)
+				if err != nil {
+					return fmt.Errorf("failed to wait for config promise: %w", err)
+				}
+				if !cfg.EnableNodePort {
+					res4.Reject(fmt.Errorf("nat ipv4: %w", MapDisabled))
+					res6.Reject(fmt.Errorf("nat ipv6: %w", MapDisabled))
+					return nil
+				}
+
+				ipv4Nat, ipv6Nat = GlobalMaps(cfg.EnableIPv4,
+					cfg.EnableIPv6, true)
+				// Maps are still created in daemon.initMaps(...) under the same circumstances
+				// so we just open them here so they can be provided to hive.
+				//
+				// TODO: Refactor ctmap gc Enable() such that it can use the map descriptors from
+				// here so we can move all nat map creation logic into here.
+				if err := ipv4Nat.Open(); err != nil {
+					return fmt.Errorf("open ipv4 nat map: %w", err)
```

Comment:
Please try to use the right capitalisation in user-facing messages, here and elsewhere in the code. Ideally I'd recommend doing the same for comments in the code as well, but I don't mind too much.

```suggestion
					return fmt.Errorf("open IPv4 NAT map: %w", err)
```